### PR TITLE
feat: add deepClone function

### PR DESCRIPTION
**Pull request by Enduriel**
Adds a deepClone function which should really be in the standard library or at the very least in MSU for a while, this is an old PR that we overlooked and I'd like for it to be included in 1.3 if possible.

### CODE REVIEW

File: msu/utils/globals.nut [hunk 1]
```diff
@@ -127,3 +127,28 @@
...
+			{
+				ret[key] <- ::MSU.deepClone(value);
+			}
+			return ret;
+		case "array":
+			ret = array(_object.len());
+			foreach (idx, value in _object)
+			{
+				ret[idx] = ::MSU.deepClone(value);
+			}
```
Comment (LordMidas) [inline msu/utils/globals.nut:148]: Use native map function: `return _object.map(@(_a) ::MSU.deepClone(_a));`

File: msu/utils/globals.nut [hunk 2]
```diff
@@ -127,3 +127,28 @@
...
+::MSU.deepClone <- function( _object )
+{
+	local ret;
+	switch (typeof _object)
+	{
+		case "table":
+			ret = {};
+			foreach (key, value in _object)
+			{
+				ret[key] <- ::MSU.deepClone(value);
```
Comment (LordMidas) [inline msu/utils/globals.nut:140]: How does cloning of tables handle delegation? Ideally we'd want a deep clone of a table with tables which has delegation set up should return a setup with the exact same delegation setup.

**Review by LordMidas — CHANGES_REQUESTED** [review]

File: msu/utils/globals.nut (hunk 2 above)
Comment (Enduriel) [reply msu/utils/globals.nut:140]: This is a good point and I agree

**Comment by Enduriel** [comment]
huh apparently the reforged branch already has a `deepClone` but it is less nice and this one is generally superior, especially with the delegation cloning.

**Review by LordMidas — APPROVED** [review]
Please edit the PR description to say `deepClone` instead of `deepEquals`.

**Review by TaroEld — APPROVED** [review]
I wonder if it works with BB objects.

**Pull request merged**